coordinates_benchmark/utils.py

```diff
@@ -0,0 +1,179 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Configuration parameters and utility functions and classes.
+"""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import os
+import logging
+import itertools
+from importlib import import_module
+import numpy as np
+from astropy.table import Table
+from astropy.coordinates import Angle
+
+TABLE_FORMAT = 'ascii.fixed_width_two_line'
+FLOAT_FORMAT_OUTPUT = '%15.10f'   # For output tables
+FLOAT_FORMAT_INPUT = '%20.15f'  # for input tables
+
+
+# Make a list of celestial conversions to check
+# We simply list all possible combinations here,
+# systems not supported by certain tools are simply skipped later
+CELESTIAL_SYSTEMS = sorted('fk5 fk4 icrs galactic ecliptic'.split())
+CELESTIAL_CONVERSIONS = itertools.product(CELESTIAL_SYSTEMS, CELESTIAL_SYSTEMS)
+CELESTIAL_CONVERSIONS = [dict(zip(['in', 'out'], _))
+                         for _ in CELESTIAL_CONVERSIONS
+                         if _[0] != _[1]]
+
+TOOLS = sorted('astropy kapteyn novas pyast palpy pyephem pyslalib pytpm'.split())
+TOOL_PAIRS = [_ for _ in itertools.product(TOOLS, TOOLS)
+              if _[0] < _[1]]
+
+
+def _import_name(name):
+    if name == 'pyast':
+        return 'starlink.Ast'
+    elif name == 'pyephem':
+        return 'ephem'
+    else:
+        return name
+
+
+def _tool_check():
+    tools = Table()
+    tools['tool_name'] = TOOLS
+    tools['import_name'] = [_import_name(_) for _ in TOOLS]
+    tools['available'] = False
+    # TODO: this is a temp workaround for this issue:
+    # https://github.com/astropy/astropy/issues/3321
+    tools['version'] = ['N/A' for _ in range(len(tools))]
+
+    for tool in tools:
+        try:
+            module = import_module(tool['import_name'])
+            tool['available'] = True
+            try:
+                version = module.__version__
+                tool['version'] = version
+            except AttributeError:
+                pass
+        except ImportError:
+            pass
+
+    return tools
+
+# Table with tools are available and what their version is
+TOOL_INFO = _tool_check()
+
+
+def select_tools(tools, include_idl=False):
+    """Select the sub-set of requested and available tools."""
+    all = list(TOOL_INFO['tool_name'])
+
+    if include_idl:
+        all.add_row(['idl', True, 'N/A'])
+
+    available = TOOL_INFO['tool_name'][TOOL_INFO['available']]
+
+    requested = tools.split(',')
+
+    if 'all' in requested:
+        requested = all
+
+    missing = set(requested).difference(available)
+    if missing:
+        logging.error('The following tools are not available: {}'.format(', '.join(missing)))
+        exit(1)
+
+    return sorted(requested)
+
+
+# TODO: get rid of this helper function!
+def our_angular_separation(lon1, lat1, lon2, lat2):
```

Comment:
This is the cause of the big changes visible in the summary ... this way to compute separations on the sky in arcsecs gives different results from the `_vicenty_dist_arcsec` function we used previously.
I'll think the results from this version are incorrect ... not sure where the precision is lost yet ... I'll fix it this afternoon.